src/reactionsystem.jl

```diff
@@ -3,19 +3,32 @@ function Catalyst.ReactionSystem(model::SBML.Model; kwargs...)  # Todo: requires
     rxs = mtk_reactions(model)
     u0map = get_u0map(model)
     parammap = get_paramap(model)
-    defs = ModelingToolkit._merge(u0map, parammap)
-    ReactionSystem(rxs,Catalyst.DEFAULT_IV,first.(u0map),first.(parammap); defaults=defs, name=gensym(:SBML), kwargs...)
+    defs = ModelingToolkit._merge(Dict(u0map), Dict(parammap))
+
+    algrules, obsrules, raterules = get_rules(model)
+    for o in obsrules
+        defs[o.lhs] = substitute(o.rhs, defs)
+        # push!(defs, o.lhs => substitute(o.rhs, defs))
+    end
+    constraints_sys = ODESystem(vcat(algrules, raterules), Catalyst.DEFAULT_IV; name = gensym(:CONSTRAINTS))
+
+    ReactionSystem(rxs, Catalyst.DEFAULT_IV, first.(u0map), first.(parammap); defaults = defs, name = gensym(:SBML),
+        observed = obsrules, constraints = constraints_sys, kwargs...)
 end
 
 """ ODESystem constructor """
-function ModelingToolkit.ODESystem(model::SBML.Model; kwargs...)
+function ModelingToolkit.ODESystem(model::SBML.Model; include_zero_odes = false, kwargs...)
     rs = ReactionSystem(model; kwargs...)
-    convert(ODESystem, rs)
+    sys = convert(ODESystem, rs; include_zero_odes = include_zero_odes)
+    ODESystem(equations(sys);
+        name = nameof(sys),
+        defaults = ModelingToolkit.get_defaults(rs),
+        continuous_events = get_events(model, rs))
 end
 
 """ Check if conversion to ReactionSystem is possible """
 function checksupport(filename::String)
-    not_implemented = ["listOfRules", "listOfConstraints", "listOfEvents"]
+    not_implemented = ["listOfRules", "listOfConstraints"]
```

Comment:
@anandijain I thought listOfRules are implemented now, but listOfEvents not.

Reply:
just fixed that

Reply:
oh i misread this. yeah events are still wip. thinking on it again ill probably split this pr up and do rules first